pkgs/swift2objc/lib/src/transformer/transform.dart

```diff
@@ -8,24 +8,55 @@ import '../ast/_core/interfaces/nestable_declaration.dart';
 import '../ast/declarations/compounds/class_declaration.dart';
 import '../ast/declarations/compounds/struct_declaration.dart';
 import '../ast/declarations/globals/globals.dart';
+import '_core/dependencies.dart';
 import '_core/unique_namer.dart';
 import 'transformers/transform_compound.dart';
 import 'transformers/transform_globals.dart';
 
 typedef TransformationMap = Map<Declaration, Declaration>;
 
-List<Declaration> transform(List<Declaration> declarations) {
+Set<Declaration> generateDependencies(Iterable<Declaration> decls, {Iterable<Declaration>? allDecls}) {
+  final dependencies = <Declaration>{};
+  final dependencyVisitor = DependencyVisitor();
+
+  var _d = decls;
+
+  while (true) {
+    final deps = _d.fold<Set<String>>(
+      {},
+      (previous, element) =>
+          previous.union(dependencyVisitor.visitDeclaration(element)));
+    final depDecls =
+      (allDecls ?? decls).where((d) => deps.contains(d.name));
```

Comment:
Names aren't unique (eg two nested types, nested in different parent types, can have the same name). Use the declaration `id` instead.

Better yet, why not just have the dependency visitor construct a `Set<Declaration>` directly, rather than constructing a `Set<String>` and then looking up the declaration from the string?

Reply:
Most of the types do not reference declarations directly (like function parameters, variable types), which therefore means that declarations need to be looked up eventually.

The lookup is done afterwards to prevent multiple lookups and just have a single lookup per visitation.

Reply:
You shouldn't ever need to look up types by name or id, after parsing is complete. The AST already has references to the declarations. Check if the `ReferredType` is a `DeclaredType`.